eclair-core/src/main/scala/fr/acinq/eclair/Eclair.scala

```diff
@@ -70,13 +70,16 @@ class EclairImpl(appKit: Kit) extends Eclair {
     (appKit.switchboard ? Peer.Connect(NodeURI.parse(uri))).mapTo[String]
   }
 
-  override def open(nodeId: PublicKey, fundingSatoshis: Long, pushMsat: Option[Long], fundingFeerateSatByte: Option[Long], flags: Option[Int]): Future[String] = {
+  override def open(nodeId: PublicKey, fundingSatoshis: Long, pushMsat: Option[Long], fundingFeerateSatByte: Option[Long], flags: Option[Int], timeout_opt: Option[Timeout]): Future[String] = {
+    // we want this to expire *before* the ask times out, otherwise the user won't get a usable response
+    val timeoutCappedAt50s_opt = timeout_opt.map(value => Timeout(value.duration.max(50 seconds)))
```

Comment:
@sstone I think that's actually not needed. Users may want a short api response, but still let time for channel opening.